src/tools/adig.c

```diff
@@ -151,15 +155,88 @@ static void print_help(void)
     "              SOA, SRV, TXT, TLSA, URI, CAA, SVCB, HTTPS\n\n");
 }
 
+static int split_rcline(char *rcline, char **rcargv, int len) {
+  int   rcargc;
+  char *last = NULL;
+
+  for (rcargc = 1, rcargv[rcargc] = strtok_r(rcline, " \n", &last);
+    rcargc < len && rcargv[rcargc];
+    rcargv[++rcargc] = strtok_r(NULL, " \n", &last))
+  {
+    DEBUGF(fprintf(stderr, "split_rcline() argc: %d, argv: %s\n", rcargc, rcargv[rcargc]));
+  }
+
+  return (rcargc);
+}
+
 static ares_bool_t read_cmdline(int argc, const char * const *argv,
-                                adig_config_t *config)
+                                adig_config_t *config, ares_bool_t is_rcfile)
 {
-  ares_getopt_state_t state;
-  int                 c;
+  ares_getopt_state_t  state;
+  int                  c;
+  char                 configdir[256];
+  char                *configdir_xdg;
+  unsigned int         cdlen = 0;
+  unsigned int         cdsize = sizeof(configdir);
+  char                *homedir;
 
   ares_getopt_init(&state, argc, argv);
   state.opterr = 0;
 
+  if (!is_rcfile) {
+
+#if defined(WIN32)
+    cdlen = (unsigned int) snprintf(configdir, cdsize, "%s", getenv("%%APPDATA%"));
+
+#elif defined(__APPLE__)
+    homedir = getenv("HOME");
+    if (homedir != NULL) {
+      cdlen = (unsigned int) snprintf(configdir, cdsize, "%s/%s/%s/%s", homedir, "Library", "Application Support", "c-ares");
+    }
+
+#else
+    configdir_xdg = getenv("XDG_CONFIG_HOME");
+
+    if (configdir_xdg == NULL) {
+      homedir = getenv("HOME");
+      if (homedir != NULL) {
+        cdlen = (unsigned int) snprintf(configdir, cdsize, "%s/%s", homedir, ".config");
+      }
+    }
+    else {
+      cdlen = (unsigned int) snprintf(configdir, cdsize, "%s", configdir_xdg);
+    }
+
```

Comment:
Think we'd need a subdirectory in both these cases as I don't think the xdg config home nor the .config directory usually just have files directly in it, there's usually a per-app directory.

Reply:
In mine there are various files without a subdirectory:

```
$ find ~/.config -maxdepth 1 -type f|wc -l
106
```

The [specification](https://specifications.freedesktop.org/basedir-spec/0.6/) says

> directory relative to which user-specific configuration files should be written

making it sound like files are fine.

But no strong opinion from me.